### PR TITLE
Improve shadows by default

### PR DESCRIPTION
**Pull request by patrikjuvonen**
This lowers shadows and lights more towards ground level by default in MTA.

#1699 worked on this, but it edited a variable which is referenced by 20 different places unrelated to shadows and lights which we want to lower, hence it's safer to go through the addresses we want to edit, and apply the new float to them only.

I agree with @botder in that we don't really need to have a setting for this because it's a general improvement and not really something you want to toggle really.

A future PR could make this float editable in Lua maybe if it's useful for any scripters.

- Ground work done by @TFP-dev in #1699 
- Lots of help and list of addresses by @botder - thank you very much for teaching and explaining

I tweaked the float a little bit and found 0.013f is a safe number and doesn't cause any flickering.

- No flickering in 16 bit color mode and/or with near/far clip distance

### CODE REVIEW

**Comment by botder** [comment]
You should add it to the **multiplayer_sa** project, because **game_sa** is meant to be a game interface library and **multiplayer_sa** is there for code modifications.

**Comment by patrikjuvonen** [comment]
> You should add it to the **multiplayer_sa** project, because **game_sa** is meant to be a game interface library and **multiplayer_sa** is there for code modifications.

Thanks, moved the code under multiplayer_sa!

File: Client/multiplayer_sa/CMultiplayerSA.cpp [hunk 1]
```diff
@@ -1498,6 +1498,29 @@ void CMultiplayerSA::InitHooks()
     MemPut((void*)0x6943AD, &fDuckingHealthThreshold);
     fDuckingHealthThreshold = 0;
 
+    // Lower the GTA shadows offset closer to ground/floor level
+    m_fShadowsOffset = 0.013f;            // GTA default = 0.06f
+    std::array<DWORD, 16> shadowAddr{
```
Comment (botder) [inline Client/multiplayer_sa/CMultiplayerSA.cpp]: Please move the array out of the function and mark it as static. Also, replace type `DWORD` with `uint32_t`.

Comment (patrikjuvonen) [reply Client/multiplayer_sa/CMultiplayerSA.cpp]: Thanks! This has been addressed.

**Review by botder — APPROVED** [review]

**Pull request merged**